src/transformStateWithClones.js

```diff
@@ -7,7 +7,25 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const resultState = { ...state };
+  const result = [];
+
+  for (const key of actions) {
+    if (key.type === 'addProperties') {
+      Object.assign(resultState, key.extraData);
+    } else if (key.type === 'removeProperties') {
+      for (const step of key.keysToRemove) {
+        delete resultState[step];
+      }
+    } else if (key.type === 'clear') {
+      for (const keys in resultState) {
+        delete resultState[keys];
+      }
```

Comment:
Instead of multiple 'if' statements, use a 'switch' statement to handle the different action types as it is more appropriate when you have a limited amount of conditions to handle. Additionally, include a 'default' case to handle any unexpected 'type' values for better error handling.